examples/Watchdog_live_view_demo.ipynb

```diff
@@ -0,0 +1,489 @@
+    "\n",
+    "            self.file_count += 1\n",
+    "            self.check_observer()\n",
+    "\n",
+    "            self.time_step.value = (self.t / self.n) * (self.file_count - 1)\n",
+    "\n",
+    "        if file_path.endswith(r\"m0.omf\"):\n",
+    "            layout = pn.Row(self.plot, self.time_step)\n",
+    "            # display one updating dmap that initially gets created with m0 file\n",
+    "            display(layout)\n",
+    "\n",
+    "        if file_path.endswith(r\".json\"):\n",
+    "            # read info.json to have the simulation time and number of files\n",
+    "            with open(file_path, \"r\") as info:\n",
+    "                self.data = json.load(info)\n",
+    "            self.t = self.data[\"t\"]\n",
+    "            self.n = self.data[\"n\"]\n",
+    "\n",
+    "    def widget(self):\n",
+    "        time_step = pn.indicators.Number(\n",
+    "            name=\"t:\",\n",
+    "            format=\"{value:.2g}\",\n",
+    "            title_size=\"10pt\",\n",
+    "            font_size=\"10pt\",\n",
+    "            default_color=\"white\",\n",
+    "        )\n",
+    "        self.time_step = time_step\n",
+    "\n",
+    "    def check_observer(self):\n",
+    "        if self.file_count >= self.n + 1:\n",
+    "            print(\"All files produced, stopping monitoring...\")\n",
+    "            self.observer.stop()\n",
+    "            self.stop_flag[0] = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The function below initialises the Watchdog Observer to monitor the desired directory.  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def main(working_directory=os.getcwd()):\n",
+    "    \"\"\"\n",
+    "    Function to plot hv plots whilst simultaneously running a Time Driver simulation\n",
+    "\n",
+    "    Parameters:\n",
+    "    working_directory: The directory we want Watchdog to monitor. Defaulted to the current working directory\n",
+    "    \"\"\"\n",
+    "\n",
+    "    # stop flag used to terminate thread when all files are produced.\n",
+    "    stop_flag = [False]\n",
+    "\n",
+    "    # Set up the observer\n",
+    "    observer = Observer()\n",
+    "    handler = DirectoryMonitorHandler(observer, stop_flag)\n",
+    "\n",
+    "    # Schedule the observer to monitor the working directory\n",
+    "    observer.schedule(handler, working_directory, recursive=True)\n",
+    "    print(f\"Monitoring: {working_directory}\")\n",
+    "\n",
+    "    # Start the observer\n",
+    "    observer.start()\n",
+    "\n",
+    "    try:\n",
+    "        while not stop_flag[0]:\n",
+    "            time.sleep(5)  # Keeps the script running\n",
+    "    except KeyboardInterrupt:\n",
+    "        print(\"Stopping monitoring...\")\n",
+    "    finally:\n",
+    "        observer.stop()\n",
+    "\n",
+    "    observer.join()"
```

Comment:
- This way of keeping the function running is not particularly elegant. Could we somehow make use of the notebook event loop?